day3/slice_time_image.py

```diff
@@ -82,12 +114,32 @@ def main():
     fname = sys.argv[1]
     # Assume the TR
     TR = 2.0
+    
     # Assume the slices were acquired even slices first, inferior to
     # superior, then odd slices, inferior to superior, where the most inferior
     # slice is index 0 and 0 is an even number.  What are the slice acquisition
     # times in seconds, where the first value is the acquisition time of slice
     # 0, the second is acquisition time of slice 1, etc?
-    slice_times = ?
+    
+    # have to load in image to know the number of slices there are
+    img = nib.load(fname)
+    num_slices = img.shape[2]
+    print "Number of slices: %s" % num_slices
+    
+    # use ceil in the next two lines in case of an odd number of slices
+    first_half = np.arange(0, np.ceil(num_slices/2.))
+    second_half = np.arange(np.ceil(num_slices/2.), num_slices)
+    slice_times = np.array([])
+    
+    for i in first_half:
+        slice_times = np.append(slice_times,first_half[i])
+        if i < len(second_half):
+            # if there are an odd number of slices
+            # len(second_half)<len(first_half)
+            # so we have to make sure that there are more slices to append
+            slice_times = np.append(slice_times,second_half[i])
```

Comment:
Nice algorithm ...

There are fancier ones, but yours is fairly easy to follow and well commented.